### PR TITLE
OCT-629 Actions bar affiliations changes

### PR DESCRIPTION
**Pull request by florin-holhos**
The purpose of this PR was to update Action Bar so that is clear for co-authors they need to confirm their affiliations first, before approving the publication.

---

### Acceptance Criteria:

As per [OCT-629](https://jiscdev.atlassian.net/browse/OCT-629)

---

### Tests:

Updated api and e2e tests
image
image

---

### Screenshots:
Corresponding user
image
image

Co-author
image
image
image

[OCT-629]: https://jiscdev.atlassian.net/browse/OCT-629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Harry-Jisc** [comment]
Happy to approve, working fine locally for myself. @shaunajisc @finlay-jisc want to also give it a look over, some large changes, might be good to have another pair of eyes?

**Review by Harry-Jisc — APPROVED** [review]

**Review by shaunajisc — APPROVED** [review]
Nice one! Looks good and it's working as expected locally

**Review by finlay-jisc — APPROVED** [review]
Also works for me, 1 small suggestion

File: api/src/components/coauthor/controller.ts [hunk 1]
```diff
@@ -277,13 +277,22 @@ export const updateConfirmation = async (
...
+
         // Is the coauthor actually a coauthor of this publication
-        if (!publication.coAuthors.some((coAuthor) => coAuthor.linkedUser === event.user.id)) {
+        if (!coAuthor) {
             return response.json(403, {
                 message: 'You are not a co-author of this publication.'
             });
         }
 
+        // check if coauthor confirmed his affiliations
```
[the remark this answers is not in the document]
Comment (florin-holhos) [reply api/src/components/coauthor/controller.ts:289]: indeed

**Review by Harry-Jisc — APPROVED** [review]

**Pull request merged**